back/api/admin.py

```diff
@@ -87,11 +87,22 @@ class OrderItemInline(admin.StackedInline):
 class OrderAdmin(admin.OSMGeoAdmin):
     change_form_template = 'admin/api/order_change_form.html'
     inlines = [OrderItemInline]
+    search_fields = ['id', 'title', 'client__identity__first_name', 'client__identity__last_name', 'client__email']
+    list_display = ['id', 'title', 'client_first_name', 'client_last_name', 'client_email', 'date_ordered']
     raw_id_fields = ['client']
     map_template = 'admin/gis/osm.html'
     ordering = ['-id']
     actions = ['quote']
-    list_filter = ['status',]
+    list_filter = ['status', 'date_ordered']
+
+    def client_first_name(self, order):
+        return order.client.identity.first_name
+
+    def client_last_name(self, order):
+        return order.client.identity.last_name
+
+    def client_email(self, order):
+        return order.client.email
```

Comment:
I do not get what this is for?

Reply:
This is the way admin.py works. By default, it will only show properties of current model, but here, in the AdminOrder instance, I want to filter and sort the list of orders by linked models so i need to create those properties by saying AdminOrder.client_last_name = Order.client.identity.last_name

Reply:
👍